src/ComputeSharp.D2D1/Shaders/Interop/D2D1ShaderCompiler.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System;
+using System.Buffers;
+using System.Text;
+using ComputeSharp.D2D1.Exceptions;
+#if !NET6_0_OR_GREATER
+using ComputeSharp.D2D1.NetStandard.System.Text;
+#endif
+using ComputeSharp.D2D1.Shaders.Translation;
+using TerraFX.Interop.DirectX;
+using TerraFX.Interop.Windows;
+
+namespace ComputeSharp.D2D1.Interop;
+
+/// <summary>
+/// Provides methods to manually compile D2D1 shaders.
+/// </summary>
+public static class D2D1ShaderCompiler
+{
+    /// <summary>
+    /// Compiles a new HLSL shader from the input source code.
+    /// </summary>
+    /// <param name="source">The HLSL source code to compile.</param>
+    /// <param name="entryPoint">The entry point of the shader being compiled.</param>
+    /// <param name="shaderProfile">The shader profile to use to compile the shader.</param>
+    /// <param name="options">The compiler options to use to compile the shader.</param>
+    /// <returns>The bytecode for the compiled shader.</returns>
+    /// <exception cref="FxcCompilationException">Thrown if the compilation fails.</exception>
+    public static ReadOnlyMemory<byte> Compile(
```

Comment:
Can we also get something along the lines of `D2D1ShaderInfo` out of the `Compile` methods? Is it readily available?

Reply:
We cannot, no, as that requires additional work. The info retrieved by that API are generated by a call to [D3DReflect](https://docs.microsoft.com/en-us/windows/win32/api/d3dcompiler/nf-d3dcompiler-d3dreflect) on the compiled bytecode. Then from that `ID3D11ShaderReflection` instance, they get the description which contains the stuff returned in `D2D1ShaderInfo`. It's not "free" from just having a shader bytecode.